repose-aggregator/core/core-lib/src/main/java/org/openrepose/powerfilter/PowerFilter.java

```diff
@@ -383,6 +384,21 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
             mutableHttpResponse.sendError(HttpServletResponse.SC_BAD_GATEWAY, "Error processing request");
             mutableHttpResponse.setLastException(ex);
         } finally {
+            try {
+                boolean isValidMethod = false;
+                for (HttpComponentFactory hcf : HttpComponentFactory.values()) {
+                    if (hcf.name().equalsIgnoreCase(mutableHttpRequest.getMethod())) {
+                        isValidMethod = true;
+                    }
+                }
+                if (!isValidMethod) {
+                    throw new Exception();
+                }
+            } catch (Exception e) {
+                LOG.debug("{}:{} -- Invalid HTTP method requested: {}", clusterId, nodeId, mutableHttpRequest.getMethod(), e);
+                mutableHttpResponse.sendError(HttpServletResponse.SC_BAD_REQUEST, "Error processing request");
+                mutableHttpResponse.setLastException(e);
+            }
```

Comment:
You don't want this in the finally block. It needs to run before the filter chain is run.